tools/autograd/derivatives.yaml

```diff
@@ -1791,6 +1792,7 @@
 - name: hardshrink_backward(Tensor grad_out, Tensor self, Scalar lambd) -> Tensor
   grad_out: hardshrink_backward(grad, self, lambd)
   self: zeros_like(grad)
+  result: at::where((self_p > lambd).logical_or(self_p < -lambd), grad_out_t, zeros_like(result))
```

Comment:
No need to generate a full Tensor of 0s here?
Also `self_t` is not involved here?

Reply:
> No need to generate a full Tensor of 0s here?

Fixed

> Also self_t is not involved here?

Yeah the values of self are only used to compute the condition for where. Otherwise hardshrink backward is constant wrt self.